otter/util/http.py

```diff
@@ -543,3 +545,16 @@ def try_json_with_keys(maybe_json_error, keys):
         return None
     else:
         return get_in(keys, error_body, None)
+
+
+def text(thing):
```

Comment:
The purpose of this function, as I understand it, is to provide a _temporary_ "lenient" text converter for code which is sloppily bandying about bytes objects.  This makes sense, but it should have a future-looking plan for how to deal with it.  My earlier comment was to suggest that it should be implemented for py3, but a better future-looking plan would be to just start emitting deprecation warnings for bytes to ensure that everyone using APIs marked as having a `text` converter are properly passing text.  (Consequently, this should have a more descriptive name for that role, like `lenient_ascii_text`.)

For what it's worth, the purpose of `six.u` is to be used only with string _literals_, not with string input from other code.  Early versions of python 3 did not support the `u''` prefix, which is what `u()` was designed to mimic; it was a way you could say `u("someval")` and get an ascii-only unicode string on both 2 and 3.  Since every version of py3 that you should even consider supporting these days is capable of the `u""` syntax, an explicit `thing.decode("ascii")` would be more explicit.